apps/web/app/quantity-leaf-slider.tsx

```diff
@@ -1,12 +1,35 @@
+import { useMemo } from 'react'
+
 export function LeafsQuantitySlider({
-  leafsQuantity,
-  setLeafsQuantity,
   label,
+  limit,
+  onChange,
+  value,
 }: {
-  leafsQuantity: number
-  setLeafsQuantity: (value: React.SetStateAction<number>) => void
   label: string
+  limit: number
+  onChange: (value: React.SetStateAction<number>) => void
+  value: number
 }): JSX.Element {
+  const sliderMarks = useMemo(() => {
```

Comment:
```
  const sliderMarks = useMemo(() => {
      if (limit <= 6) {
        return Array.from({ length: limit - 1 }, (_, i) => i + 2)
      }

      const spans = [2]
      const step = (limit - 2) / 5

      for (let i = 1; i <= 5; i++) {
        spans.push(Math.round(2 + step * i))
      }

      return spans
  }, [limit])
  ```

Reply:
no need for generateSliderMarks function

Reply:
small example that is much easier to follow imho

```
const max = 100
const markerCount = 6
const firstMark = 2


const a = [ firstMark, ...Array.from({length: markerCount - 1}, (_, i) => firstMark + Math.round((i + 1)*(max - firstMark)/(markerCount-1)))]

console.log(a)
```

Reply:
could be simpler if firstmark is just added and the other marks are not influenced by the first Mark